### ISSUE TITLE
Validate and ensure HIP Graph / CUDA Graph replay compatibility

### ISSUE THREAD

**Issue by sunway513** [sunway513/rocm-trace-lite#15]
## Context

Almost all production inference workloads use HIP Graph (CUDA Graph on ROCm) for kernel replay. This is the primary use case for rocm-trace-lite. If graph replay bypasses our HSA queue intercept, the tool is effectively useless for production profiling.

## How HIP Graph works at HSA level

1. **Capture phase**: Application dispatches kernels normally — these go through `hsa_queue_create` and our intercept callback sees them
2. **Replay phase**: The captured graph is replayed via `hipGraphLaunch` which submits pre-built AQL packets directly to the GPU queue, potentially bypassing the intercept callback entirely

## Risk

If graph replay uses:
- **`hsa_amd_queue_intercept_create` queues**: our callback fires — we're safe
- **Internal/private queues or direct ring buffer writes**: our callback is bypassed — we see nothing during replay
- **Pre-built packets with pre-assigned signals**: our signal replacement in `queue_intercept_cb` may corrupt the graph's internal signal chain

### 1. Trace a HIP Graph workload and check capture vs replay
```python
import torch

x = torch.randn(256, 256, device="cuda")
stream = torch.cuda.Stream()

# Warmup
with torch.cuda.stream(stream):
    y = x @ x
stream.synchronize()

# Capture
g = torch.cuda.CUDAGraph()
with torch.cuda.graph(g, stream=stream):
    y = x @ x

# Replay 100 times
for _ in range(100):
    g.replay()
stream.synchronize()
```

Expected: if working, should see ~101 GEMM dispatches (1 warmup + 100 replays).
If broken: only 1-2 dispatches (capture phase only).

### 2. Check if graph replay corrupts execution
Run the same workload with and without rpd_lite. Compare output tensors.

### 3. Investigate HSA-level graph replay mechanism
- Does `hipGraphLaunch` use `hsa_amd_queue_intercept_create` queues?
- Does ROCm's HIP graph implementation use a separate internal queue?
- Check ROCm source: `hipamd/src/hip_graph.cpp` and `hip_graph_internal.hpp`

### 4. Check how rtg_tracer handles this
The original rtg_tracer in rocmProfileData may or may not handle graph replay. Check if it has specific graph support.

### A. Graph replay goes through intercepted queues (best case)
- No code changes needed
- Add HIP Graph tests to CI

### B. Graph replay bypasses intercept (likely case)
Options:
1. **Intercept `hipGraphLaunch`** at HIP level (needs roctracer — defeats our purpose)
2. **Intercept at HSA queue submission level** — if replay uses `hsa_queue_add_write_index`, we may be able to intercept via a different HSA table hook
3. **Use `hsa_amd_queue_intercept_register` on the graph's internal queue** — if we can identify which queue the graph uses
4. **Accept the limitation** — document that graph replay is not traced, only capture-phase dispatches. Users can disable graph (`enforce_eager=True`) for profiling sessions.

### C. Graph replay crashes with rpd_lite loaded (worst case)
- Our signal replacement in queue_intercept_cb may corrupt pre-built graph packets
- Need to detect graph context and pass packets through unmodified
- Check `HSA_PACKET_HEADER_TYPE` for graph-specific packet types

## Test matrix

| Scenario | What to check |
|----------|--------------|
| PyTorch CUDAGraph capture+replay | Dispatch count matches expected |
| ATOM with cudagraph_capture_sizes | Full model inference traces correctly |
| vLLM with --enforce-eager=False | Graph-heavy serving traces correctly |
| Graph replay output correctness | No corruption from signal interception |
| Multiple graph instances | No cross-graph interference |

## Priority

**P0** — this blocks production use. Most ATOM/vLLM deployments use HIP Graph by default.

## Files
- `src/hsa_intercept.cpp`: `queue_intercept_cb()` — may need graph-aware logic
- `tests/test_gpu_multithread.py`: add HIP Graph test cases
